app/api/groups/recent_activity.go

```diff
@@ -39,11 +41,12 @@ func (srv *Service) getRecentActivity(w http.ResponseWriter, r *http.Request) se
 			 IF(user_strings.idLanguage IS NULL, default_strings.sTitle, user_strings.sTitle) AS Item__String__sTitle`).
 		Where("users_answers.idItem IN ?",
 			srv.Store.ItemAncestors().All().DescendantsOf(itemID).Select("idItemChild").SubQuery()).
-		Where("users_answers.sType='Submission'")
-	query = srv.Store.Items().JoinStrings(user, query)
-	query = srv.Store.Items().KeepItemsVisibleBy(user, query)
-	query = srv.Store.GroupAncestors().KeepUsersThatAreDescendantsOf(groupID, query)
-	query = query.Order("users_answers.sSubmissionDate DESC, users_answers.ID")
+		Where("users_answers.sType='Submission'").
+		SoThat(items.OnlyVisibleBy(user)).
+		SoThat(users.OnlyDescendantsOf(groupID)).
```

Comment:
But so you apply a `OnlyDescendantsOf` filter here on something not related with users...  so I assume it is for joining with the userID of user_answers.... ok ....
But with a step back again, having `SoThat(users.OnlyDescendantsOf(groupID)).` when we just to chain a function is quite complex/ugly/unreadable for something simple...
Mmh related with another comment I did about the fact your functor what not really related with `users` or even directly with `items` (while `Visible(..)` using it does... shouldn't this functor moved away in way that they can be chained... Actually, most of our fct on DB return a DB which is just a gorm query, but couldn't we imagine that it is a query on which `OnlyDescendantsOf` is implemented? ...
... I drifted from the initial comment on this line, sorry 😁

Reply:
For instance, you could still have the concept of `DB` with initial method such as `Table()` or `Insert` and next to it `Table()` would return a kind of `ComposableQuery` on which you can apply `Select`, `Where`, ... but also all kind of higher level filter such as `.KeepOnlyDescendantsOf(user, joining_key)` or `.JoinItemStrings()` or `.WhereItemVisibleBy(user)` 
🤔 mixing both kind of fct is maybe not great but there should be a way to do it cleanly in Go (?)

Reply:
I think, this is the way... soThat...

Reply:
> But so you apply a `OnlyDescendantsOf` filter here on something not related with users... so I assume it is for joining with the userID of user_answers.... ok ....
But with a step back again, having `SoThat(users.OnlyDescendantsOf(groupID)).` when we just to chain a function is quite complex/ugly/unreadable for something simple...

```.Given(items.JoinUsers).SoThat(users.AreDescendantsOf(groupID))```

Reply:
`Given` is a nice synonym, I believe

Reply:
I (try to) explain another way of solving the problem and that is your only answer? 😕

Reply:
That's another thread. Here that's not about naming.